src/Umbraco.Web/Editors/UserEditorAuthorizationHelper.cs

```diff
@@ -79,6 +79,13 @@ public UserEditorAuthorizationHelper(IContentService contentService, IMediaServi
             {
                 var savingGroupAliases = userGroupAliases.ToArray();
 
+                // As we know the current user is not admin, it is only allowed to use groups that the user do have themselves.
```

Comment:
Doesn't seem like the comment here matches up with the logic below.  I.e. it sounds like the intention is to not allow the assignment of groups that the user isn't themselves part of, but actually it's only restricting the setting of the admin group.

Reply:
You are right, the comment is not updated.. The comment match my first implementation but that was not correct. Ill fix, thanks

Reply:
Actually, the logic in the UI seems to match the original comment.. I will look into what is up and down..